eng/common/TestResources/New-TestResources.ps1

```diff
@@ -343,20 +310,25 @@ try {
         }
     }
 
-    # Get test application OID from ID if not already provided.
-    if ($TestApplicationId -and !$TestApplicationOid) {
-        $testServicePrincipal = Retry {
-            Get-AzADServicePrincipal -ApplicationId $TestApplicationId
-        }
+    # Determine the Azure context that the script is running in.
+    $context = Get-AzContext;
 
-        if ($testServicePrincipal -and $testServicePrincipal.Id) {
-            $script:TestApplicationOid = $testServicePrincipal.Id
+    # Make sure the provisioner OID is set so we can pass it through to the deployment.
+    $provisionerApplicationOid = if (!$ProvisionerApplicationId) {
+        if ($context.Account.Type -eq 'User') {
+            $user = Get-AzADUser -UserPrincipalName $context.Account.Id
+            $user.Id
+        } elseif ($context.Account.Type -eq 'ServicePrincipal') {
+            $sp = Get-AzADServicePrincipal -ApplicationId $context.Account.Id
+            $sp.Id
+        } else {
+            Write-Warning "Getting the OID for provisioner type '$($context.Account.Type)' is not supported and will not be passed to deployments (seldom required)."
         }
+    } else {
+        $sp = Get-AzADServicePrincipal -ApplicationId $ProvisionerApplicationId
+        $sp.Id
```

Comment:
Is this right? The `Id` property is the `Oid`?

Reply:
Yes. There's a separate `ApplicationId` that is what we generally think of the `Id`. I did some tests specifically for this case - comparing with the much easier-to-understand output of the CLI - to be sure. 🙂